version: "3.8"

services:
  db:
    container_name: phototom_db
    image: postgres:14-alpine
    restart: always
    user: postgres # importante definir o usuário
    volumes:
      - pgdata:/var/lib/postgresql/data
    environment:
      - LC_ALL=C.UTF-8
      - POSTGRES_USER=postgres # usuário padrão
      - POSTGRES_PASSWORD=postgres # senha padrão
      - POSTGRES_DB=phototom_db # necessário porque foi configurado assim no settings
    ports:
      - 5431:5432 # repare na porta externa 5431
    networks:
      - phototom-network

  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: phototom-api
    hostname: app
    stdin_open: true
    expose:
      - '8000'
    volumes:
      - /core/.env:/phototom/.env
    command: bash -c "gunicorn core.wsgi:application -b 0.0.0.0:8000"
    depends_on:
      - db
    networks:
      - phototom-network

  nginx:
    container_name: phototom_nginx
    image: nginx
    hostname: nginx
    ports:
      - '80:8000'
    volumes:
      - ./docker/config/nginx/:/etc/nginx/conf.d/
    depends_on:
      - app
    networks:
      - phototom-network

volumes:
  pgdata: # mesmo nome do volume externo definido na linha 10
  pgadmin:


networks:
  phototom-network:
